src/result.ts

```diff
@@ -1,18 +1,61 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Upstream docs about the result type https://bytecodealliance.github.io/jco/wit-type-representations.html#result-result
+
 const prototype = {
+    /**
+     * Returns `true` if the result is successful, `false` otherwise
+     * @example.
+     * Result.ok(123).isOk() // true
+     * @example
+     * Result.err('error').isOk() // false
+     */
+    isOk,
+    /**
+     * Returns `true` if the result is an error, `false` otherwise
+     * @example
+     * Result.err('error').isOk() // false
+     * @example.
+     * Result.ok(123).isOk() // true
+     */
+    isErr,
+    /**
+     * Returns the successful value of the result or throws the error value.
+     * This should be used when returning from a component function that expects to receive a result.
+     * See https://bytecodealliance.github.io/jco/wit-type-representations.html#results-in-context-function-return-values for more details.
+     * @example Returns the payload of a successful result.
+     * Result.ok(123).unlift() // 123
+     * @example Throws the payload of a failed result.
+     * Result.err('error').unlift() // throws 'error'
+     */
+    unlift,
```

Comment:
i fell it will  be confusing for users to have both unwrap and unflit, is there a string reason to have both?

Reply:
In most other cases throwing a non-error object is not nice and having the symmetry with unwrapErr (when it comes to throwing a TypeError instead of an unknown object) is nice.
But I agree that it can be a bit confusing